### PR TITLE
Fix Mono so libil2cpp with debugger compiles on UWP and Xbox One

### PR DESCRIPTION
**Pull request by TautvydasZilys**
These changes already landed to IL2CPP repo. 

Katana: https://katana.bf.unity3d.com/projects/Mono%20Runtime%20and%20Classlibs/builders?boo_branch=unity-trunk&cecil_branch=unity-master&il2cpp_branch=master&krait-signal-handler_branch=master&mono_branch=fix-uwp-and-xbox-one-build-for-debugger&mono-build-deps_branch=default&unityscript_branch=unity-trunk

### CODE REVIEW

File: mono/utils/mono-threads-windows.c [hunk 1]
```diff
@@ -29,13 +29,12 @@ mono_threads_suspend_begin_async_suspend (MonoThreadInfo *info, gboolean interru
 	HANDLE handle;
 	DWORD result;
 
-	handle = OpenThread (THREAD_ALL_ACCESS, FALSE, id);
+	handle = info->native_handle;
```
Comment (joncham) [inline mono/utils/mono-threads-windows.c]: Does the OpenThread call need done? I forget my Win32 but doesn't it convert a pseudu handle to a real handle or something for later API calls?

**Comment by joncham** [comment]
All good. I'd like to upstream most of this as it all seems like general Windows goodness?

File: mono/utils/mono-threads-windows.c (hunk 1 above)
Comment (TautvydasZilys) [reply mono/utils/mono-threads-windows.c]: I use DuplicateHandle in mono_threads_suspend_register in order to convert a pseudohandle to a real handle. OpenThread is unnecessary.

**Comment by TautvydasZilys** [comment]
@joncham Sure. I'm not sure which changes they'd be against taking or whether anything is against their conventions, though.

**Comment by joshpeterson** [comment]
I'll take these changes, along with some I have, upstream next week.

**Pull request merged**